src/EventListener/SwitchUserListener.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\BackendUser;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Http\Event\SwitchUserEvent;
+
+class SwitchUserListener
+{
+    protected $logger;
+    protected $tokenStorage;
+
+    public function __construct(LoggerInterface $logger, TokenStorageInterface $tokenStorage)
+    {
+        $this->logger = $logger;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    public function onSwitchUser(SwitchUserEvent $event)
+    {
+        /** @var BackendUser $user */
+        $user = $this->tokenStorage->getToken()->getUser();
+
+        /** @var BackendUser $targetUser */
+        $targetUser = $event->getTargetUser();
+
+        $this->logger->info(sprintf("User %s has switched to user %s.", $user->username, $targetUser->username));
```

Comment:
The only purpose of this listener is to log the action? If that's a replacement of an existing log, we should log with a `ContaoContext` to make it appear in the backend log.